### ISSUE TITLE
Support jump lists

### ISSUE THREAD

**Issue by bsclifton** [brave/browser-laptop#6094]
**Did you search for similar issues before submitting this one?**
YEs

**Describe the issue you encountered:**
Similar to https://github.com/brave/browser-laptop/issues/3024, Windows should show a jump list

**Expected behavior:**
When Brave is pinned to the taskbar in Windows, you should be able to right click it and get a jump list of common actions, similar to what you do with other Windows apps

Here are some great docs with information about how we can do this (see `User Tasks (Windows)`):
http://electron.atom.io/docs/tutorial/desktop-environment-integration/

(should be as easy as making a few calls to the `app.setUserTasks` API)

For now, we might only be able to offer a "New window" option (since we don't have a concept of a private window yet).

Here's an example with a site pinned to taskbar from Internet Explorer (see the "Tasks" section)
![screen shot 2016-12-08 at 11 25 08 am](https://cloud.githubusercontent.com/assets/4733304/21022208/0639e042-bd39-11e6-8ae2-476940835d2d.png)

- Platform (Win7, 8, 10? macOS? Linux distro?):
Windows

**Comment by Liunkae** [comment]
Related: #5141 (New Private Window in jump list)

**Comment by ShinyStar** [comment]
Hello!
I'd like to work on this issue and I have a few questions.
I've been looking at cmdLine.js and I haven't found any flags to open a new window from the command line. Am i missing it or should I add it? 
And, if I did add it, would "--new-window" be ok as the flag? Should I make just one branch for both things, or one for adding the flag and another for the jump list?

**Comment by NumDeP** [comment]
@flamsmark don't you think it's a good idea to stick a feature/tor label on this issue and to additionally display something similar to the bottom inferences: 

- Open New Window
- New Amnesiac Window 
- (TAB) Tor Amnesiac Browsing

It will probably defeat the purpose for some users to insecurely open a new window and then toggle TAB afterwards if there intention is to TAB.

**Comment by flamsmark** [comment]
@NumDeP This feature doesn't depend on new Tor modes for implementation. When new Tor features are implemented, we can add them as additional options here and in the equivalent places for other docks/launchers. No reason to block this right now.

**Comment by NumDeP** [comment]
I'm sorry I wasn't aware that there was going to be more features implemented on top of what the Tor browser already has. I was under the assumption @flamsmark that it was literally going to include a couple of Toggles, one in the Shield area, one in the menu and one in the jump lists which is which is why I presented the above comments.

That being said, you're statement doesn't indicate that the jump lists issue can't be accelerated, considering that users that have a prior intention of accessing the internet over Tor would prefer to connect to it much quicker rather than passing through multiple notices and yet another toggle gives a reason for this issue to be tied with a feature/tor label, correct :)

**Comment by bsclifton** [comment]
@NumDeP this feature doesn't have anything to do with Tor (although, having a `new tor tab` would be cool)

@ShinyStar I'd love to help you work on this 😄 The docs linked to in the original post should be enough to help get started.  Please note: we use Muon, a fork of electron... but much of the code is the same 😄 

Maybe start by getting a jump list showing and then we can go from there.  The full chromium command line flags [can be found here](https://peter.sh/experiments/chromium-command-line-switches/) (although, we don't support all of them).  Let me know if you have any more questions or if you want a review

**Comment by ShinyStar** [comment]
@bsclifton Thanks a lot for the help! I already have the jump list showing. As the post, said, its just a simple call to app.setUserTasks(). My problem is, however, getting it to open the new window because there isn't (or I haven't found) a way to open a new window by calling the program with arguments to do so. I thought I could implement that first and then apply it here, and I'm looking into it now. :)

**Comment by ShinyStar** [comment]
@bsclifton Hi! I did [this](https://github.com/ShinyStar/browser-laptop/commit/a69158dc11d1a330a4fe43c7af7d737387494153) to add the command line argument and jump list but since I'm calling Brave from the development environment it results in just Electron so I can't really test if the jump list works, I have only checked that the command line argument works by modifying npm start. How could I generate an installer to make sure that it really works?
